manifests/init.pp

```diff
@@ -18,6 +14,12 @@
 #  don't have READ access to the default 'incident' table. Note that you
 #  can turn the ServiceNow credentials validation off by setting this
 #  parameter to the empty string ''.
+# @param [Optional[String[1]]] user
+#   The username of the account
```

Comment:
Ah, should we specify that one set of credentials must be provided? I liked what you had in https://github.com/puppetlabs/puppetlabs-servicenow_cmdb_integration/blob/main/manifests/init.pp#L7-L14